### ISSUE TITLE
Support importing ES modules into .NET

### ISSUE THREAD

**Issue by jasongin** [microsoft/node-api-dotnet#235]
Currently the `NodejsEnvironment.Import()` method supports importing CommonJS modules. But attempting to import an ES module results in an error:

```
Unexpected token 'export'
```

[Node.js supports ES modules.](https://nodejs.org/api/esm.html) We need to figure out how to import them in the `libnode` embedding scenario.

**Comment by jasongin** [comment]
This can be accomplished now by invoking the async dynamic `import()` operator via the `RunScript()` method, then awaiting the resulting promise:

```C#
JSPromise esModulePromise = (JSPromise)JSValue.RunScript("import('path/to/esm-module')");
JSValue esModule = await esModulePromise.AsTask();
```

I'll add another API to simplify it though.

**Comment by ceztko** [comment]
Hello, sorry for disturbing you: even though I fought a lot with CommonJS vs ESModule semantics, I have a relatively fresh knowledge of Node.js and possibly I may misunderstand the sense of this issue.

It is documented that native addons can be loaded through [`require()`](https://nodejs.org/api/addons.html), hence implicitly forcing the module to be CommonJS. Well, it turned out it's not the only way to load a native addon in NodeJS: the official [`process.dlopen(module, exports)`](https://nodejs.org/api/process.html#processdlopenmodule-filename-flags) function can be used to load any shared library, even those that are not named `.node` (eg. `.dll`, `.so`, `.dylib`). With `process.dlopen` and [`url.fileURLToPath`](https://nodejs.org/api/url.html#urlfileurltopathurl) I was able to make a [bootstrapper](https://github.com/ceztko/CodeBinder-TestCodeGen/blob/f50c697e8ac2e7207fca4f44929be00685e4a6ad/SampleLibraryMTS/NAPISampleLibrary.mts#L53) that allowed me to load the shared library and keep the whole module following ESModule semantics. I am pretty much sure you could make an alternative bootstrapper that makes the modules built with `node-api-dotnet` to be proper ESModule(s) and be able to use them in your examples like the following:

```js
import { Example } from 'aot-npm-package'
const result = Example.hello('.NET AOT');
```

Interestingly enough, `process.dlopen` was not well known even among some Node.js [maintainers](https://github.com/nodejs/node/issues/46650#event-8822160616).

**Comment by jasongin** [comment]
@ceztko Interesting idea that is maybe worth exploring further. But it is an entirely separate issue from this one, and not the highest priority right now.

**Comment by ceztko** [comment]
Ok, as said I'm sorry: there was the possibility I could misunderstand the issue. I stumbled on this one with a keyword search because we were trying to load a Native AOT compiled module built with `node-api-dotnet` as ESModule, and while this is possible the experience feels a little slack. Since I already made native addons that are ESModule compliant I wanted to inform you.

**Comment by jasongin** [comment]
I opened #239. Let's move any further discussion about @ceztko's suggestion over there.

**Comment by jasongin** [comment]
Fixed in version 0.6.

Use the new [ImportAsync()](https://github.com/microsoft/node-api-dotnet/blob/5cce1cc202a5c3f0463e4e05076d89bde6f8c836/src/NodeApi/Runtime/NodejsEnvironment.cs#L374) method with `esModule: true`.

(I just noticed that method is missing doc-comments. I'll fix that later.)
